boston-crime-data/content.md

```diff
@@ -0,0 +1,379 @@
+from matplotlib import pyplot
+
+crime_data = pd.read_csv('crime_data.csv', encoding='windows-1252')
+crime_and_weather_data = pd.read_csv('crime_and_weather_data.csv', encoding='windows-1252')
+count_cutoff = 5000
+
+
+print('HOURLY COUNT')
+print('Total crime')
+hourly_count = crime_data.Hour.value_counts()
+pyplot.bar(hourly_count.index, hourly_count)
+pyplot.show()
+type_count = crime_data.Type.value_counts()
+for e in type_count.index:
+    if type_count[e] > count_cutoff:
+        print(e)
+        specific_hourly_count = crime_data[crime_data.Type == e].Hour.value_counts()
+        pyplot.bar(specific_hourly_count.index, specific_hourly_count)
+        pyplot.show()
+
+
+print(' ')
+print('WEEKLY COUNT')
+print('Total crime')
+crime_data.Weekday = crime_data.Weekday.replace(['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday',
+                                                 'Sunday'], [1, 2, 3, 4, 5, 6, 7])
+daily_count = crime_data.Weekday.value_counts()
+pyplot.bar(daily_count.index, daily_count)
+pyplot.show()
+type_count = crime_data.Type.value_counts()
+for e in type_count.index:
+    if type_count[e] > count_cutoff:
+        print(e)
+        specific_daily_count = crime_data[crime_data.Type == e].Weekday.value_counts()
+        pyplot.bar(specific_daily_count.index, specific_daily_count)
+        pyplot.show()
+
+print(' ')
+print('MONTHLY COUNT')
+print('Total crime')
+crime_data['Date'] = pd.to_datetime(crime_data['Date'])
+crime_data['Monthday'] = crime_data.Date.dt.day
+monthday_count = crime_data.Monthday.value_counts()
+pyplot.bar(monthday_count.index, monthday_count)
+pyplot.show()
+type_count = crime_data.Type.value_counts()
+for e in type_count.index:
+    if type_count[e] > count_cutoff:
+        print(e)
+        specific_monthday_count = crime_data[crime_data.Type == e].Monthday.value_counts()
+        pyplot.bar(specific_monthday_count.index, specific_monthday_count)
+        pyplot.show()
+
+print(' ')
+print('YEARLY COUNT')
+print('Total crime')
+crime_data['Date'] = pd.to_datetime(crime_data['Date'])
+crime_data['Month'] = crime_data.Date.dt.month
+monthly_count = crime_data.Month.value_counts()
+pyplot.bar(monthly_count.index, monthly_count)
+pyplot.show()
+type_count = crime_data.Type.value_counts()
+for e in type_count.index:
+    if type_count[e] > count_cutoff:
+        print(e)
+        specific_monthly_count = crime_data[crime_data.Type == e].Month.value_counts()
+        pyplot.bar(specific_monthly_count.index, specific_monthly_count)
+        pyplot.show()
+```
+
+## Foracasting with RNN
+Now we build and train a small recurrent neural network. Make a file called `hourly_model.py` where we will use the hourly
+data we have prepared. Firstly, there are a few dependencies to import. Keras helps us build the network, sklearn/scikitlearn 
+is used to standardize the data and measure error, and of course mflux_ai/Mflux to log models and data over several runs.
+```python
+import pandas as pd
```

Comment:
Optimize/order the imports by using Pycharm's built-in "Optimize imports" functionality